pkg/enqueue/Tests/Client/Driver/RabbitMqDriverTest.php

```diff
@@ -62,6 +62,43 @@ public function testShouldCreateQueueWithMaxPriorityArgument()
         $this->assertSame(['x-max-priority' => 4], $queue->getArguments());
     }
 
+    public function testShouldCreateLazyQueueifisLazySetinConfig()
+    {
+        $context = $this->createContextMock();
+        $context
+            ->expects($this->once())
+            ->method('createQueue')
+            ->willReturn($this->createQueue('aname'))
+        ;
+
+        $config = Config::create(
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            [
+                'dsn' => getenv('RABBITMQ_AMQP_DSN'),
+            ],
+            ['rabbit_mq_lazy_queues' => ['aname']]
+        );
+
+        $driver = $this->createDriver(
+            $context,
+            // presumably we pass in some isLazy check here
+            $config,
+            new RouteCollection([])
+        );
+
+        /** @var AmqpQueue $queue */
+        $queue = $driver->createQueue('aname');
+        $queueArguments = $queue->getArguments();
+        $this->assertSame($queueArguments['x-queue-mode'], 'lazy');
+        $this->assertSame($queueArguments['x-max-priority'], 4);
```

Comment:
Add a check for a `Queue` that has a different name, since we're here, proving that `x-queue-mode` will not appear in queues not listed in `$config` as lazy.

```php
$queue = $driver->createQueue('bname');
$queueArguments = $queue->getArguments();
$this->assertFalse(isset($queueArguments['x-queue-mode']));
$this->assertSame($queueArguments['x-max-priority'], 4);
```

Remove the comment `// presumably we pass in some isLazy check here` just to keep it clean and in my opinion you're done :smile: 

We might add an info about this functionality in the docs, if you or someone else finds a suitable place :smile:

Reply:
I'm happy to add this but I'm not sure where it makes sense to add this in the docs. Would you accept this as a separate PR for the docs instead?

Reply:
@mrchrisadams Don't duplicate the test :smile: Just add the four lines I've posted above at the end of this test :smile:

Reply:
@mrchrisadams 

> I'm happy to add this but I'm not sure where it makes sense to add this in the docs. Would you accept this as a separate PR for the docs instead?

Yes, I'll follow this PR with opening an issue, so we can take our time adding proper documentation. I believe it's a functionality that you need, so docs "can wait".

Moreover, if makasim will want to wait before publishing new version - for whatever reason, even if simply to not publish for each and every small functionality - I can show you how to override repositories so you can use your enqueue build immediately, before a new version is officially out.

See https://getcomposer.org/doc/05-repositories.md#loading-a-package-from-a-vcs-repository